### DISCUSSION TITLE
Only updating certain fields from a form

### DISCUSSION THREAD

**Discussion by aantn** [Budibase/budibase#1728] [Help]
Hi,
I'm using a form to edit an object from TableA which has a one-to-many relationship named SomeField with TableB. 

I've set the form for TableA to hide SomeField so that the user can't edit it and I write the form's contents back to TableA using a "Save Row" action. However, the "Save Row" action doesn't work unless I explicitly set SomeField to be "{{ Repeater.TableA.SomeField }}". This is problematic because the relationship might have changed in the interim if a new row was added to TableB.

What is the correct way to write back changes to TableA without modifying it's relationships with another table?

**Comment by mike12345567** [comment]
Hi @aantn the team had a quick discussion around this, we're going to make a small change for this, we are going to make it possible to send up a row with only it partially filled, when patching a row our API will fill in the rest of the row based on what is currently stored in the database.

I should note however a point about how internal tables work in Budibase, they are based on CouchDB which has a concept of a `_rev` key that is attached to each document, this revision key tells Couch if the data you're sending up is the latest version or not, so that when you try to update a row it can fail if you have a stale version of the row. In this case the request will throw a `409 Document Conflict` error.

Under most circumstances this shouldn't be an issue, when updating a row the form will fetch the latest version of the row and pre-populate the form with that data, meaning that there is only a chance for the data to become stale between opening a screen that contains a form and the actual save of the form.

Hopefully this makes sense, if it doesn't let me know, also would be great to know what you think of this, if you have a scenario where you expect a row to have been updated between it being fetched by the form and it being saved i'd love to hear it!

**Reply by aantn** [reply]
Thanks, that solution sounds good. 

Can you elaborate how this works with relationships? How are relationships stored in CouchDB by Budibase?

I'm encountering what seems to be a separate bug where saving a row that has relationships always fails with the generic "An Error Occurred" message. Would be happy to send some logs if you let me know where to gather them from.

**Reply by mike12345567** [reply]
Relationships are a little bit different in Budibase, for the most part they won't really be affected by this system as they are stored outside all rows. In Budibase the implementation of relationships is through joining documents, we essentially store another document that describes the relationship between two different rows. This technically means that relationships don't quite follow the rules around how revisions are handled, right now the relationships with a row could be stale and it would not error.

Interesting, yes that does sound like a bug of some sort, it would be great if you could open dev tools in your browser and look at the network requests, its likely a row API call is failing and it'll be provided a bit more of an indepth error. There may also be errors in the browser console.

Edit: if you're interested in a deeper dive on how relationships are represented in Budibase [this blog post](https://danielwertheim.se/couchdb-many-to-many-relations/) covers the subject of many -> many relationships in Couch very well, but beware, it requires some pretty indepth knowledge about CouchDBs architecture!
